File: C/6.适配器模式/src/Mp4Player.c
/**
 * @file Mp4Player.c
 * @author {Layne} ({shu_huanglei@163.com})
 * @brief
 * @version 0.1
 * @date 2020-07-25
 *
 * @copyright Copyright (c) 2020
 *
 */

#include <stdio.h>

#include "AdvancedMediaPlayerStruct.h"
#include "ctools.h"

static void play_vlc(struct AdvancedMediaPlayer* advanced_media_player, const char* file_name) {}

static void play_mp4(struct AdvancedMediaPlayer* advanced_media_player, const char* file_name) {
    printf("Playing mp4 file. Name: %s\n", file_name);
}

struct AdvancedMediaPlayer* mp4_player_create(void) {
    struct AdvancedMediaPlayer* player = (struct AdvancedMediaPlayer*)calloc(1, sizeof(struct AdvancedMediaPlayer));
    player->play_mp4 = play_mp4;
    player->play_vlc = play_vlc;

    return player;
}

File: Objective-C/6.AdapterPattern/Mp4Player.m
//
//  Mp4Player.m
//  6.AdapterPattern
//
//  Created by Layne on 2020/10/22.
//

#import "Mp4Player.h"

@implementation Mp4Player

- (void)playMp4:(nonnull NSString *)fileName {
    NSLog(@"Playing mp4 file. Name: %@", fileName);
}

- (void)playVlc:(nonnull NSString *)fileName {
}

@end
